shardingsphere-test/shardingsphere-rewrite-test/src/test/java/org/apache/shardingsphere/sharding/rewrite/parameterized/engine/AbstractSQLRewriterParameterizedTest.java

```diff
@@ -120,6 +122,9 @@ public final void assertRewrite() throws IOException {
                 ? Collections.singletonList(((GenericSQLRewriteResult) sqlRewriteResult).getSqlRewriteUnit()) : (((RouteSQLRewriteResult) sqlRewriteResult).getSqlRewriteUnits()).values();
     }
     
+    protected void mockDataSource(final Map<String, DataSource> dataSources) throws SQLException {
```

Comment:
@tuichenchuxin Please make mockDataSource abstract.

Reply:
Thank you. Done.